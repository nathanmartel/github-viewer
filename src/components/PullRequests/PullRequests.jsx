import React from 'react';
import PropTypes from 'prop-types';
import PullRequest from '../PullRequest/PullRequest';


const PullRequests = ({ pullRequests }) => {
  const pullRequestElements = pullRequests.map(pullRequest => (
    pullRequest.id && 
      <li key={pullRequest.id}>
        <PullRequest {...pullRequest} />
      </li>
  ));
  
  return (
    <ul>
      {pullRequestElements}
    </ul>
  );
};

PullRequests.propTypes = {
  pullRequests: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    number: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  })).isRequired
};

export default PullRequests;
