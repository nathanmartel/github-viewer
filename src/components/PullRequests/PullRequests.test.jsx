import React from 'react';
import { shallow } from 'enzyme';
import PullRequests from './PullRequests';

describe('PullRequests component', () => {
  it('renders PullRequests', () => {
    const pullRequests = [{ 
      id: 1,
      number: 5,
      title: 'Add this',
      state: 'open',
      url: 'http://www.hello.com'
    }, 
    { 
      id: 2,
      number: 6,
      title: 'Subtract that',
      state: 'closed',
      url: 'http://www.hello.com'
    }];
    const wrapper = shallow(<PullRequests pullRequests={pullRequests} />);
    expect(wrapper).toMatchSnapshot();
  });
});
