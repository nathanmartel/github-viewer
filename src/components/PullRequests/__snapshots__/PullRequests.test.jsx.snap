// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PullRequests component renders PullRequests 1`] = `
<ul>
  <li
    key="1"
  >
    <PullRequest
      id={1}
      number={5}
      state="open"
      title="Add this"
      url="http://www.hello.com"
    />
  </li>
  <li
    key="2"
  >
    <PullRequest
      id={2}
      number={6}
      state="closed"
      title="Subtract that"
      url="http://www.hello.com"
    />
  </li>
</ul>
`;
